src/Signals/Signal.m

```diff
@@ -164,6 +164,70 @@ function clearData(sObj)
             newSobj.setData( dataBlock );
         end
         
+        function dataBlockResampled = resampleToFsHz( sObj, dataBlock, srcFsHz )
+            %resampleToFsHz  This method resamples a data block
+            %               to match the signal's sampling rate.
+            %               Resampling is perforrmed via linear
+            %               interpolation without decimation.
+            %
+            %USAGE:
+            %   resample = sObj.resampleToFsHz( dataBlock, srcFsHz )
+            %
+            %INPUT ARGUMENTS:
+            %  dataBlock : The data block to resample with signal's FsHz
+            %  srcFsHz : the original sampling rate of the data block
+            %  
+            [rows, ~] = size(dataBlock);
+            x = 0 : 1 / srcFsHz : (rows-1) / srcFsHz;
+            xq = 0 : 1 / sObj.FsHz : (rows-1) / srcFsHz;
+            dataBlockResampled = interp1( x, dataBlock, xq );

```

Comment:
@hagenw , Since the we're resampling with a uniform rate, resample() would be more straightforward. But it's only defined in the signal processing toolbox.  I thought I shouldn't assume everyone will have the toolbox.